src/Query/ResultPrinters/PrefixParameterProcessor.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace SMW\Query\ResultPrinters;
+
+use SMWQuery as Query;
+use SMW\Query\Language\NamespaceDescription;
+
+trait PrefixParameterProcessor {
+
+	private bool $mixedResults;
+	private string $prefix;
+
+	public function setPrefix( string $prefix ) {
+		$this->prefix = $prefix;
+	}
+
+	public function setQuery( Query $query ) {
+		$this->mixedResults = $this->determineMixedResults( $query );
+	}
+
+	private function determineMixedResults( Query $query ) : bool {
+		// this is a basic implementation, possibly to be expanded,
+		// to guess whether result entries are expected to have
+		// homogeneous or mixed prefixes
+		return !( $query->getDescription() instanceof NamespaceDescription );
+	}
+
+	public function useLongText( bool $isSubject ) : bool {
+		$prefix = $this->prefix;
+
+		if ( $prefix === 'all'
+			|| ( $prefix === 'subject' && $isSubject )
+			|| ( $prefix === 'auto' && $isSubject && $this->mixedResults ) ) {
```

Comment:
This class has various design issues.

Code will blow up is setQuery was not called first.

setQuery does work that is not always needed, since mixedResults is only read in some cases

Actually it is not a class, it is a trait, which have been dubbed "compiler-assisted copy-paste"

ValueTextsBuilder is still requiring a complex object in its constructor. Instead you could inject the boolean, which seems to be all it needs, or wrap figuring out the boolean into a class and then inject said class. You could turn this trait into a class.

Reply:
redesigned using a class. Regarding the boolean, I'm not sure that a boolean is all we need. There could be other ways in the method PrefixParameterProcessor->getMixedResults to determine whether the query is expected to return mixed namespaces. For now I've found one, but there could be more I'm not aware of. (for this reason this method at least by principle needs to evaluate the query itself.)